openedx/tests/xblock_integration/test_done.py

```diff
@@ -0,0 +1,139 @@
+"""
+This tests that the completion XBlock correctly stores state. This
+is a fairly simple XBlock, and a correspondingly simple test suite.
+"""
+
+import json
+import unittest
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+
+from lms.djangoapps.courseware.tests.helpers import LoginEnrollmentTestCase
+from lms.djangoapps.courseware.tests.factories import GlobalStaffFactory
+from lms.djangoapps.lms_xblock.runtime import quote_slashes
+
+
+class TestDone(SharedModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Simple tests for the completion XBlock. We set up a page with two
+    of the block, make sure the page renders, toggle them a few times,
+    make sure they've toggled, and reconfirm the page renders.
+    """
+    STUDENTS = [
+        {'email': 'view@test.com', 'password': 'foo'},
+        {'email': 'view2@test.com', 'password': 'foo'}
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Create a page with two of the XBlock on it
+        """
+        # Nose runs setUpClass methods even if a class decorator says to skip
+        # the class: https://github.com/nose-devs/nose/issues/946
+        # So, skip the test class here if we are not in the LMS.
+        if settings.ROOT_URLCONF != 'lms.urls':
+            raise unittest.SkipTest('Test only valid in lms')
+
+        super(TestDone, cls).setUpClass()
+        cls.course = CourseFactory.create(
+            display_name='Done_Test_Course'
+        )
+        with cls.store.bulk_operations(cls.course.id, emit_signals=False):
+            cls.chapter = ItemFactory.create(
+                parent=cls.course, display_name='Overview'
+            )
+            cls.section = ItemFactory.create(
```

Comment:
If category is not specified, I believe it defaults to "chapter". Therefore, if you want the items being created to match your intended structure, you should specify the category for section and unit.